web-ng/src/app/components/feature-details/feature-details.component.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Input, OnInit, OnDestroy, NgZone } from '@angular/core';
+import { Layer } from '../../shared/models/layer.model';
+import { getPinImageSource } from '../map/ground-pin';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { ConfirmationDialogComponent } from '../confirmation-dialog/confirmation-dialog.component';
+import { MatDialog } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { DataStoreService } from '../../services/data-store/data-store.service';
+import { NavigationService } from './../../services/router/router.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-feature-details',
+  templateUrl: './feature-details.component.html',
+  styleUrls: ['./feature-details.component.scss'],
+})
+export class FeatureDetailsComponent implements OnInit, OnDestroy {
+  @Input() layer?: Layer;
+  @Input() actionsType: LayerListItemActionsType =
+    LayerListItemActionsType.MENU;
+  projectId?: string | null;
```

Comment:
Can we just use `projectId?` (without |null)

Reply:
Hey @gino-m , w.r.t #459, | null with ? as in the router service we have projectId?: string | null and paramap returns an observable of string | null type.